src/java.base/share/classes/java/util/TimeZone.java

```diff
@@ -615,24 +616,63 @@ private static TimeZone getTimeZone(String ID, boolean fallback) {
     /**
      * Gets the available IDs according to the given time zone offset in milliseconds.
      *
+     * @apiNote Consider using {@link #availableIDs(int)} which returns
+     * a stream of the available time zone IDs according to the given offset.
+     *
      * @param rawOffset the given time zone GMT offset in milliseconds.
      * @return an array of IDs, where the time zone for that ID has
      * the specified GMT offset. For example, "America/Phoenix" and "America/Denver"
      * both have GMT-07:00, but differ in daylight saving behavior.
      * @see #getRawOffset()
+     * @see #availableIDs(int)
      */
     public static synchronized String[] getAvailableIDs(int rawOffset) {
         return ZoneInfo.getAvailableIDs(rawOffset);
     }
 
     /**
-     * Gets all the available IDs supported.
-     * @return an array of IDs.
+     * {@return an array of the available IDs supported}
+     *
+     * @apiNote Consider using {@link #availableIDs()} which returns
+     * a stream of the available time zone IDs.
+     *
+     * @see #availableIDs()
      */
     public static synchronized String[] getAvailableIDs() {
         return ZoneInfo.getAvailableIDs();
     }
 
+    /**
+     * Gets the available IDs according to the given time zone offset in milliseconds.
+     *
+     * @implNote Unlike {@link #getAvailableIDs(int)}, this method does
+     * not create a copy of the {@code TimeZone} IDs array.
+     *
+     * @param rawOffset the given time zone GMT offset in milliseconds.
+     * @return a stream of IDs, where the time zone for that ID has
+     * the specified GMT offset. For example, "America/Phoenix" and "America/Denver"
+     * both have GMT-07:00, but differ in daylight saving behavior.
+     * @see #getRawOffset()
+     * @see #getAvailableIDs(int)
+     * @since 25
+     */
+    public static synchronized Stream<String> availableIDs(int rawOffset) {
```

Comment:
Why are these new methods synchronized? I think the streams and the spliterators are late-binding, so the synchronization does not synchronize the access to the underlying arrays. synchronized modifier also does not appear in Javadoc and is not part of the API surface, so it can be safely removed.

Reply:
Yep. They are not needed - https://bugs.openjdk.org/browse/JDK-8291027

Reply:
Right, it can be removed. Actually, I think `getAvailableIDs()` and `getAvailableIDs(int)` also do not need to be synchronized. The internal IDs are eagerly initialized in the static block of `ZoneInfoFile.java` and are effectively final and not modified.

Reply:
Yep; their synchronized are incorrect. The object monitor lock should have been on `ZoneInfoFile.class` to be effective.

Reply:
Related PR waiting for review - https://github.com/openjdk/jdk/pull/17441